### PR TITLE
codespelunker: init at 1.0.0

### PR DESCRIPTION
**Pull request by viraptor**
###### Description of changes

Initial build

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/codespelunker/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    hash = "sha256-NN/78paePdvYHQ4J2aQu56PvEciOXY8DxHd4ajfVCFU=";
+  };
+
+  vendorHash = null;
+
+  meta = with lib; {
+    description = "Command line codespelunker or code search";
+    homepage = "https://github.com/boyter/cs";
+    license = with licenses; [ mit unlicense ];
+    platforms = platforms.all;
```
Comment (emilylange) [inline pkgs/development/tools/codespelunker/default.nix:19]:
```suggestion
```

File: pkgs/development/tools/codespelunker/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "boyter";
+    repo = "cs";
+    rev = "v${version}";
+    hash = "sha256-NN/78paePdvYHQ4J2aQu56PvEciOXY8DxHd4ajfVCFU=";
+  };
+
+  vendorHash = null;
+
+  meta = with lib; {
+    description = "Command line codespelunker or code search";
```
Comment (emilylange) [inline pkgs/development/tools/codespelunker/default.nix:16]:
```suggestion
    description = "A command line search tool";
```

File: pkgs/development/tools/codespelunker/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://github.com/boyter/cs";
+    license = with licenses; [ mit unlicense ];
+    platforms = platforms.all;
+    maintainers = [ maintainers.viraptor ];
```
Comment (emilylange) [inline pkgs/development/tools/codespelunker/default.nix:20]: Nit

```suggestion
    maintainers = with maintainers; [ viraptor ];
```

File: pkgs/development/tools/codespelunker/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
```
Comment (emilylange) [inline pkgs/development/tools/codespelunker/default.nix:2]: Nit

```suggestion
{ lib, buildGoModule, fetchFromGitHub }:

buildGoModule rec {
```

**Review by emilylange — CHANGES_REQUESTED** [review]

**Review by emilylange — APPROVED** [review]

File: pkgs/development/tools/codespelunker/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchFromGitHub {
+    owner = "boyter";
+    repo = "cs";
+    rev = "v${version}";
+    hash = "sha256-NN/78paePdvYHQ4J2aQu56PvEciOXY8DxHd4ajfVCFU=";
+  };
+
+  vendorHash = null;
+
```
Comment (figsoda) [inline pkgs/development/tools/codespelunker/default.nix:15]: this removes the extra `csperf` and `parser` binaries which doesn't seem to be part of the main program

```suggestion

  subPackages = [ "." ];

```

File: pkgs/development/tools/codespelunker/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  meta = with lib; {
+    description = "A command code search tool";
+    homepage = "https://github.com/boyter/cs";
+    license = with licenses; [ mit unlicense ];
+    maintainers = with maintainers; [ viraptor ];
+  };
```
Comment (figsoda) [inline pkgs/development/tools/codespelunker/default.nix:21]:
```suggestion
    mainProgram = "cs";
  };
```

**Review by figsoda — APPROVED** [review]

**Pull request merged**